core-web/libs/sdk/client/src/lib/client/content/methods/get-collection/get-collection.ts

```diff
@@ -0,0 +1,285 @@
+import { Equals } from '../../../../query-builder/lucene-syntax';
+import { QueryBuilder } from '../../../../query-builder/sdk-query-builder';
+import { CONTENT_API_URL } from '../../shared/const';
+import { GetCollectionResponse, QueryBuilderCallback, SortByArray } from '../../shared/types';
+import { sanitizeQueryForContentType } from '../../shared/utils';
+
+/**
+ * 'GetCollection' Is a Typescript class that provides the ability build a query to get a collection of content.
+ *
+ * @export
+ * @class GetCollection
+ */
+export class GetCollection {
+    #page = 1;
+    #limit = 10;
+    #depth = 0;
+    #render = false;
+    #sortBy?: SortByArray;
+    #contentType: string;
+    #defaultQuery: Equals;
+    #query?: Equals;
+    #rawQuery?: string;
+
+    private serverUrl: string;
+    private requestOptions: Omit<RequestInit, 'body' | 'method'>;
+
+    /**
+     * This method returns the sort query in this format: field order, field order, ...
+     *
+     * @readonly
+     * @private
+     * @memberof GetCollection
+     */
+    private get sort() {
+        return this.#sortBy?.map((sort) => `${sort.field} ${sort.order}`).join(',');
+    }
+
+    /**
+     * This method returns the offset of the content to fetch
+     *
+     * @readonly
+     * @private
+     * @memberof GetCollection
+     */
+    private get offset() {
+        // This could end in an empty response
+        return this.#limit * (this.#page - 1);
+    }
+
+    /**
+     * This method returns the url to fetch the content
+     *
+     * @readonly
+     * @private
+     * @memberof GetCollection
+     */
+    private get url() {
+        return `${this.serverUrl}${CONTENT_API_URL}`;
+    }
+
+    /**
+     * This method returns the current query built
+     *
+     * @readonly
+     * @private
+     * @memberof GetCollection
+     */
+    private get currentQuery() {
+        return this.#query ?? this.#defaultQuery;
+    }
+
+    constructor(
+        requestOptions: Omit<RequestInit, 'body' | 'method'>,
+        serverUrl: string,
+        contentType: string
+    ) {
+        this.requestOptions = requestOptions;
+        this.serverUrl = serverUrl;
+
+        this.#contentType = contentType;
+
+        // We need to build the default query with the contentType field
+        this.#defaultQuery = new QueryBuilder().field('contentType').equals(this.#contentType);
```

Comment:
What about if we throw an error if the user don't pass the query? Or maybe I am not understanding what this is for.